From: Chad Dupuis <cdupuis@redhat.com>
Date: Fri, 3 Sep 2010 21:39:58 -0400
Subject: [scsi] qla2xxx: stop firmware before doing init firmware
Message-id: <20100903213958.2767.8850.sendpatchset@localhost.localdomain>
Patchwork-id: 28143
O-Subject: [RHEL 5.6 PATCH 1/9] qla2xxx: Stop firmware before doing init
	firmware.
Bugzilla: 619814
RH-Acked-by: Rob Evers <revers@redhat.com>

Bugzilla
--------

Bug 619814 (https://bugzilla.redhat.com/show_bug.cgi?id=619814)

Upstream Status
---------------

scsi-misc commit id 14e303d98bcfe4a6075407b531b1df2f237deb28

Description
-----------

>From 38187795802ce65b772739059b02e2f8db06f268 Mon Sep 17 00:00:00 2001
From: Lalit Chandivade <lalit.chandivade@qlogic.com>
Date: Thu, 17 Jun 2010 14:09:20 +0530
Subject: [PATCH] qla2xxx: Stop firmware before doing init firmware.

If BIOS is enabled then drivers init firmware fails since
BIOS has done the init once.

diff --git a/drivers/scsi/qla2xxx/qla_init.c b/drivers/scsi/qla2xxx/qla_init.c
index cd26aac..19b1d75 100644
--- a/drivers/scsi/qla2xxx/qla_init.c
+++ b/drivers/scsi/qla2xxx/qla_init.c
@@ -1068,8 +1068,11 @@ qla2x00_setup_chip(scsi_qla_host_t *ha)
 
 	/* Load firmware sequences */
 	rval = ha->isp_ops->load_risc(ha, &srisc_address);
-	if (IS_QLA82XX(ha))
+	if (IS_QLA82XX(ha)) {
+		if (rval == QLA_SUCCESS)
+			qla2x00_stop_firmware(ha);
 		goto enable_82xx_npiv;
+	}
 
 	if (rval == QLA_SUCCESS) {
 		DEBUG(printk("scsi(%ld): Verifying Checksum of loaded RISC "
